fletcher/_algorithms.py

```diff
@@ -261,3 +281,187 @@ def all_op(arr: Union[pa.ChunkedArray, pa.Array], skipna: bool) -> bool:
         return _all_op_nonnull(len(arr), arr.buffers()[1])
     # skipna is not relevant in the Pandas behaviour
     return _all_op(len(arr), *arr.buffers())
+
+
+def np_reduce_op(
+    npop: Callable,
+    arr: Union[pa.ChunkedArray, pa.Array],
+    skipna: bool = True,
+    identity: Optional[int] = None,
+):
+    """Use numpy operations to provide a reduction."""
+    if not skipna and arr.null_count > 0:
+        return pa.NA
+    elif isinstance(arr, pa.ChunkedArray):
+        valid_chunks = [
+            chunk for chunk in arr.iterchunks() if chunk.null_count != len(chunk)
+        ]
+        results = pa.array(
+            [
+                np_reduce_op(npop, chunk, skipna=skipna, identity=identity)
+                for chunk in valid_chunks
+            ]
+        )
+        return np_reduce_op(npop, results, skipna=skipna, identity=identity)
+    else:
+        if len(arr) == 0:
+            if identity is None:
+                raise ValueError("zero-size reduction on operation with no identity")
+            else:
+                return identity
+        np_arr = _extract_data_buffer_as_np_array(arr)
+        if arr.null_count > 0:
+            mask = extract_isnull_bytemap(arr)
+            return npop(np_arr[~mask])
+        else:
+            return npop(np_arr)
+
+
+sum_op = partial(np_reduce_op, np.sum, identity=0)
+max_op = partial(np_reduce_op, np.amax)
+min_op = partial(np_reduce_op, np.amin)
+prod_op = partial(np_reduce_op, np.prod, identity=1)
+
+
+def pd_nanop(nanop: Callable, arr: Union[pa.ChunkedArray, pa.Array], skipna: bool):
+    """Use pandas.core.nanops to provide a reduction."""
+    if isinstance(arr, pa.ChunkedArray):
+        data = pa.concat_arrays(arr.iterchunks())
+    else:
+        data = arr
+    np_arr = _extract_data_buffer_as_np_array(data)
+    mask = extract_isnull_bytemap(data)
+
+    return nanop(np_arr, skipna=skipna, mask=mask)
+
+
+std_op = partial(pd_nanop, nanops.nanstd)
+skew_op = partial(pd_nanop, nanops.nanskew)
+kurt_op = partial(pd_nanop, nanops.nankurt)
+var_op = partial(pd_nanop, nanops.nanvar)
+median_op = partial(pd_nanop, nanops.nanmedian)
+
+
+def _in_chunk_offsets(
```

Comment:
Didn't we implement this somewhere before?

Reply:
No, maybe you are referring to `_calculate_chunk_offsets` which I moved into this file?

Reply:
Ok, didn't see the removal anywhere else. The logic felt familiar but I couldn't find it right away